ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/common/worldview/DiffBasedWorldState.java

```diff
@@ -223,6 +237,14 @@ public void persist(final BlockHeader blockHeader) {
       stateUpdater
           .getWorldStateTransaction()
           .put(TRIE_BRANCH_STORAGE, WORLD_ROOT_HASH_KEY, calculatedRootHash.toArrayUnsafe());
+
+      stateUpdater
+          .getWorldStateTransaction()
+          .put(
+              TRIE_BRANCH_STORAGE,
+              WORLD_BLOCK_NUMBER_KEY,
+              Long.toHexString(blockHeader == null ? 0L : blockHeader.getNumber())
```

Comment:
same as the other one

I think you can use something like that to decode and encode in the database

encode to byte[] before writing into the database
 >  byte[] bytes = Bytes.ofUnsignedLong(100000).toArrayUnsafe();

decode from the database
> Bytes.wrap(bytes.toArrayUnsafe()).toLong();